### PR TITLE
Ifaddr pr

### PR DESCRIPTION
**Pull request by mcrowson**
A continuation of pr #120 with maintainers comments addressed. 

Ran unit tests locally and am getting one failure. Not familiar enough with the lib to know if this is an issue
```python
ERROR: test_lots_of_names (test_zeroconf.Names)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/path/to/libpython-zeroconf/test_zeroconf.py", line 345, in test_lots_of_names
    s.sendto(packet, 0, (r._MDNS_ADDR, r._MDNS_PORT))
OSError: [Errno 40] Message too long
```

### CODE REVIEW

**Comment by mcrowson** [comment]
One change from the original PR is for the network mask. Depending on how many addresses are available, the prefix could be 32, 24, or sometimes 16. I was personally seeing 24 as the prefix for my en0 and not loading the interface correctly. So I just ensured that we're looking at a string (IPv6 gives back tuples with ipaddr) and then check for IPv4 formatting with the four parts like the original PR did.

File: requirements-dev.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ flake8-blind-except
 # Upper bound because of https://github.com/public/flake8-import-order/issues/42
 flake8-import-order>=0.4.0, <0.6.0
 # See setup.py comment for why this version is restricted
```
Comment (jstasiak) [inline requirements-dev.txt]: This comment won't be necessary anymore

File: zeroconf.py [hunk 2]
```diff
@@ -1617,20 +1614,20 @@ def find(cls, zc=None, timeout=5, interfaces=InterfaceChoice.All):
...
+def get_all_addresses():
     return list(set(
-        addr['addr']
-        for iface in netifaces.interfaces()
-        for addr in netifaces.ifaddresses(iface).get(address_family, [])
-        if addr.get('netmask') != HOST_ONLY_NETWORK_MASK
+        addr.ip
+        for iface in ifaddr.get_adapters()
+        for addr in iface.ips
+        if type(addr.ip) is str and len(addr.ip.split(".")) == 4
```
Comment (jstasiak) [inline zeroconf.py]: This change removes host-only interface skipping introduced in b5e9e944e6f3c990862b3b03831bb988579ed340, that check needs to stay (with modifications, naturally).

**Comment by jstasiak** [comment]
Thank you, merged in 7c0500ee1986.

**Pull request closed without merging**